### PR TITLE
Add Mixin axiom

### PR DESCRIPTION
**Pull request by KernelDeimos**

### CODE REVIEW

File: src/foam/core/Mixin.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    },
+    'flags',
+    'path',
+    ['priority', 200]
+  ],
+
+  methods: [
+    function installInClass(cls) {
+      var m = this.__context__.lookup(this.path);
+      if ( ! m ) throw 'No such interface or trait: ' + this.path;
```
Comment (kgrgreer) [inline src/foam/core/Mixin.js]: "No such mixing." or "No such model."

File: src/foam/core/Mixin.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  `,
+
+  properties: [
+    {
+      name: 'name',
+      getter: function() { return 'mixin_' + this.path; }
+    },
+    'flags',
+    'path',
+    ['priority', 200]
```
Comment (kgrgreer) [inline src/foam/core/Mixin.js:23]: What is this used for?

File: src/foam/core/Mixin.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    properties behave as if they were defined on the model itself.
+  `,
+
+  properties: [
+    {
+      name: 'name',
+      getter: function() { return 'mixin_' + this.path; }
+    },
+    'flags',
+    'path',
```
Comment (kgrgreer) [inline src/foam/core/Mixin.js:22]: What is this used for?

File: src/foam/core/Mixin.js (hunk 2 above)
Comment (KernelDeimos) [reply src/foam/core/Mixin.js:23]: I saw priority on the implements mixin and I assume it ensures this axiom will override all existing methods and properties

File: src/foam/core/Mixin.js (hunk 3 above)
Comment (KernelDeimos) [reply src/foam/core/Mixin.js:22]: which one - path? That's the string that's passed into the AxiomArray

File: src/foam/core/Mixin.js (hunk 1 above)
Comment (KernelDeimos) [reply src/foam/core/Mixin.js]: Fixed

**Pull request merged**